### PR TITLE
update details screen

### PR DESCRIPTION
**Pull request by khrystyna-sb**
Screenshot 2021-12-09 at 13 39 59
Screenshot 2021-12-09 at 13 40 12
Screenshot 2021-12-09 at 13 42 48

### CODE REVIEW

File: Countries/DetailsViewController.swift [hunk 1]
```diff
@@ -9,51 +10,180 @@ import UIKit
...
+        view.translatesAutoresizingMaskIntoConstraints = false
+        return view
+    }()
+    
+    private let headerView: DetailsVCHeader = {
+        let view = DetailsVCHeader()
+        view.contentMode = .scaleAspectFit
+        
+        if UIDevice.current.userInterfaceIdiom == .phone {
+        view.backgroundColor = Constants.backgroundColor
```
Comment (Globalika) [inline Countries/DetailsViewController.swift]: add tabulation

File: Countries/DetailsViewController.swift [hunk 2]
```diff
@@ -9,51 +10,180 @@ import UIKit
...
+            defaultView.bottomAnchor.constraint(equalTo: view.bottomAnchor, constant: -Constants.bottomAnchorConstant)
         ])
     }
     
+    private func configureFlag() {
+        
+        guard let country = country else { return }
+        
+        let layer = CALayer()
+        layer.cornerRadius = 5
```
Comment (Globalika) [inline Countries/DetailsViewController.swift]: move 5 to constants

File: Countries/DetailsViewController.swift [hunk 3]
```diff
@@ -63,24 +193,57 @@ class DetailsViewController: UIViewController {
...
+            let imageView = UIImageView()
+            imageView.contentMode = .scaleAspectFit
+            
+            if i % 2 == 0 {
+                let image = UIImage(named: "ellipse1")
+                imageView.image = image
+            } else {
+                let image = UIImage(named: "ellipse2")
+                imageView.image = image
+            }
```
Comment (Globalika) [inline Countries/DetailsViewController.swift]: images and image names are constants so move them to constant struct

File: Countries/DetailsViewController.swift [hunk 4]
```diff
@@ -63,24 +193,57 @@ class DetailsViewController: UIViewController {
...
+            
+            let vectorImage = UIImageView()
+            vectorImage.image = UIImage(named: "vector")
+            vectorImage.contentMode = .scaleAspectFit
+            
+            let verticalStackView = UIStackView()
+            verticalStackView.axis = .vertical
+            verticalStackView.alignment = .leading
+            verticalStackView.distribution = .fillEqually
+            verticalStackView.spacing = 10
```
Comment (Globalika) [inline Countries/DetailsViewController.swift]: saa constants

File: Countries/Headers/DetailsVCHeader.swift [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+        stackView.axis = .vertical
+        stackView.distribution = .fillEqually
+        stackView.spacing = Constants.spacing
+        stackView.translatesAutoresizingMaskIntoConstraints = false
+        return stackView
+    }()
+    
+    var earthView: UIImageView = {
+        let imageView = UIImageView()
+        let image = UIImage(named: "earth")
```
Comment (Globalika) [inline Countries/Headers/DetailsVCHeader.swift]: saa

File: Countries/Headers/DetailsVCHeader.swift [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+        imageView.image = image
+        imageView.contentMode = .scaleAspectFit
+        return imageView
+    }()
+    
+   private let flagsView: UIImageView = {
+        let imageView = UIImageView()
+        let image = UIImage(named: "wing")
```
Comment (Globalika) [inline Countries/Headers/DetailsVCHeader.swift]: saa

**Review by Globalika — CHANGES_REQUESTED** [review]
i left you comments. please resolve them

File: Countries/DetailsViewController.swift [hunk 7]
```diff
@@ -213,15 +215,15 @@ class DetailsViewController: UIViewController {
             imageView.contentMode = .scaleAspectFit
             
             if i % 2 == 0 {
-                let image = UIImage(named: "ellipse1")
+                let image = UIImage(named: Constants.redBallImageName)
                 imageView.image = image
             } else {
-                let image = UIImage(named: "ellipse2")
+                let image = UIImage(named: Constants.greenBallImageName)
```
Comment (Globalika) [inline Countries/DetailsViewController.swift]: both UIImage and image name are constants. please do like this: static let redBallImage = UIImage(named: "ellipse1")

File: Countries/Headers/DetailsVCHeader.swift [hunk 8]
```diff
@@ -14,6 +14,8 @@ class DetailsVCHeader: UIView {
     
     private enum Constants {
         static let spacing: CGFloat = -60.0
+        static let earthImageName = "earth"
+        static let wingImageName = "wing"
```
Comment (Globalika) [inline Countries/Headers/DetailsVCHeader.swift]: saa

**Review by Globalika — CHANGES_REQUESTED** [review]

**Review by Globalika — APPROVED** [review]

File: Countries/Headers/DetailsVCHeader.swift [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+        imageView.contentMode = .scaleAspectFit
+        return imageView
+    }()
+    
+   private let flagsView: UIImageView = {
+        let imageView = UIImageView()
+       imageView.image = Constants.wingImage
+        imageView.contentMode = .scaleAspectFit
+        return imageView
+    }()
```
Comment (apolwork) [inline Countries/Headers/DetailsVCHeader.swift]: Alignment

File: Countries/DetailsViewController.swift [hunk 10]
```diff
@@ -9,51 +10,182 @@ import UIKit
...
     
-    private func setupStackView() {
-        view.addSubview(stackView)
+    private func configureContents() {
+        
+        if UIDevice.current.userInterfaceIdiom == .pad {
+            constraintsForIpad()
+        } else {
+            constraintsForIphone()
+        }
```
Comment (apolwork) [inline Countries/DetailsViewController.swift]: What is the difference between iPad and iPhone constraints?

Comment (khrystyna-sb) [reply Countries/DetailsViewController.swift]: there is containerView for iPad similar to the design

**Review by apolwork — APPROVED** [review]

**Pull request merged**